### PR TITLE
Issue 22319 pull relationships from child

### LINKED ISSUE #22319: Rest API: Pulling Relationships from child
We need to be able to pull relationships from the child (Lucene query) and get the parents, it must accept pagination so it needs to limit, offset and sort.

This method works but just via viewtool https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentTool.java#L396-L423

Parameters this endpoint should accept
- identifier of parent
- relationship field variable
- lucene condition to filter children
- limit
- offset
- orderby

### PR DESCRIPTION
**Pull request by jdotcms**
adding pull related parents

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/relationships/RelationshipsResource.java [hunk 1]
```diff
@@ -124,4 +132,60 @@ public final Response getOneSidedRelationships(
...
+            final User user = initData.getUser();
+            final Language language = WebAPILocator.getLanguageWebAPI().getLanguage(request);
+            final long langId = UtilMethods.isSet(luceneCondition)
+                    && luceneCondition.contains("languageId") ? -1 : language.getId();
+            final String tmDate = request.getSession (false) != null?
+                    (String) request.getSession (false).getAttribute("tm_date"):null;
+            final PageMode mode = PageMode.get(request);
+            final boolean editOrPreviewMode = !mode.showLive;
+            final List<Contentlet> retrievedContentlets = ContentUtils
+                    .pullRelated(relationshipFieldVariable, contentletIdentifier,
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/relationships/RelationshipsResource.java:182]: this method (or one of the other overloaded ones) can pull either children or parents.  Why should we limit this endpoint to only pulling parents?

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/curl-test/ContentResourceV1.postman_collection.json [hunk 2]
```diff
@@ -308,6 +308,1040 @@
 				}
 			},
```
Comment (nollymar) [inline dotCMS/src/curl-test/ContentResourceV1.postman_collection.json:309]: I'll include a test to pull self-related content (from both sides, parent and child). Probably, these cases are easier to test in an ITs, the code would be more legible and easy to maintain. The postman test is good for testing http responses for example, but testing heavier logic is by far easier in an IT.

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java [hunk 3]
```diff
@@ -232,4 +224,77 @@ private Contentlet getContentlet(final String inode,
...
+     * @param response - Http Response
+     * @param pullRelatedForm {@link PullRelatedForm}
+     * @return Returns empty List if no results are found
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("related")
+    public Response pullRelated(@Context final HttpServletRequest request,
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java:251]: Note to us - we need to begin returning structured objects and not just returning `javax.ws.rs.core.Response` so OpenApi can understand the return that is expected.  This is ok as is but future forward I think we need to do a bit more work here.

Also, how does this look in swagger?  Does it need some annotations?

**Review by wezell — CHANGES_REQUESTED** [review]
Check to see how the endpoint looks in the API playground and add annotations if needed.

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java:251]: @jdotcms take a look at this for needed Swagger annotations https://github.com/dotCMS/core/blob/9e7d22db567b68ea0f54ee8526bbb1a51e45a556/dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java#L1489

**Review by wezell — APPROVED** [review]
looking sweet

**Review by nollymar — APPROVED** [review]

**Pull request merged**